### PR TITLE
Typing: manager.py

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

**Comment by emcastillo** [comment]
```pytorch_pfn_extras/training/manager.py:668: error: Argument 1 to "len" has incompatible type "Union[DataLoader[Any], Iterable[Any], None]"; expected "Sized"  [arg-type]
```

File: pytorch_pfn_extras/training/manager.py [hunk 1]
```diff
@@ -652,7 +665,7 @@ def set_reporter_on_iter(engine: Engine) -> None:
 
         @self.engine.on(Events.STARTED)
         def set_training_started(engine: Engine) -> None:
-            iters_per_epoch = len(engine.state.dataloader)
+            iters_per_epoch = len(engine.state.dataloader)  # type: ignore[arg-type]
```
Comment (asi1024) [inline pytorch_pfn_extras/training/manager.py:668]: note: The actual type of `engine.state.dataloader` is `Union[DataLoader[Any], Iterable[Any], None]`, so the dataloader is not guaranteed to be `Sized`.

**Review by emcastillo — APPROVED** [review]
LGTM

**Pull request merged**